gui/default/syncthing/core/syncthingController.js

```diff
@@ -3343,6 +3343,11 @@ angular.module('syncthing.core')
                 return 'checkbox';
             }
             if (value instanceof Array) {
+                if (value.some(function (element) {
+                    return typeof element !== 'number' || typeof element !== 'string'
```

Comment:
Actually, the logic is wrong here. This needs to be `&&`, otherwise this always returns `true` for all non-empty lists.

```suggestion
                    return typeof element !== 'number' && typeof element !== 'string';
```

Or alternatively:

```suggestion
                    return ['number', 'string'].indexOf(typeof element) === -1;
```
(I'd like to use [`Array.includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes), but that is much younger than `some` and `indexOf`.)

Reply:
You're absolutely right, must have been just a little too tired when writing this.  As the list of types we can handle is unlikely to ever be extended, I prefer the simple logic variant without array meddling.